src/SOS/Strike/hostcoreclr.cpp

```diff
@@ -612,7 +617,11 @@ static HRESULT InitializeNetCoreHost()
 
     // Trust The SOS managed and dependent assemblies from the sos directory
     std::string tpaList;
-    AddFilesFromDirectoryToTpaList(sosModuleDirectory.c_str(), tpaList);
+    const char* directory = sosModuleDirectory.c_str();
+    AddFileToTpaList(directory, "System.Reflection.Metadata.dll", tpaList);
```

Comment:
Any reason we are being so picky? It's sort of curious because these some from Symstore, and they should be seen as a reference. Globbing was less secure but more resilient to changes in SOS. The moment we add a dependency, it might fall off the TPA list (although it probably would be caught by testing)

Reply:
This hardcoded list is the only way I could get everything to work across all the runtimes we host on.  3.1, 5.0 and 6.0 don't need anything from the SOS directory but 2.1 does need at least those. 

I know it is very fragile and I hope to come up with a better way. My upcoming extension work already ran into this problem and there is a lot more project dependencies that will have to be added. You would think `netstandard2.0` (all the project and reference assemblies are that) would work across all the runtimes.

Reply:
But TPA is about runtime, netstandard is about available surface area. Not sure how that would influence.

Reply:
But doesn't the framework it is built with (netstandard2.0) determine what dependencies are published?  If I changed everything to netcoreapp2.1 I don't think assemblies like System.Buffer or System.Memory would be published. 

FYI, using netcoreapp2.1 wouldn't work on the desktop Framework, I think.

Reply:
netstandard2.0 will mostly influence surface available, and publish forwarders as necessary (like the ones you just mentioned, which just reroute to SPC stuff on core).